packages/editor-ui/src/components/NodeCreator/NodeItem.vue

```diff
@@ -44,14 +69,77 @@ export default Vue.extend({
 		'nodeType',
 		'bordered',
 	],
+	data(): Data {
+		return {
+			dragging: false,
+			draggablePosition: {
+				x: -100,
+				y: -100,
+			},
+		};
+	},
 	computed: {
-		shortNodeType() {
+		shortNodeType(): string {
 			return this.$locale.shortNodeType(this.nodeType.name);
 		},
+		isTrigger (): boolean {
+			return this.nodeType.group.includes('trigger');
+		},
+		draggableStyle(): { top: string; left: string; } {
+			return {
+				top: `${this.draggablePosition.y}px`,
+				left: `${this.draggablePosition.x}px`,
+			};
+		},
 	},
-	// @ts-ignore
-	isTrigger (nodeType: INodeTypeDescription): boolean {
-		return nodeType.group.includes('trigger');
+	mounted() {
+		/**
+		 * Workaround for firefox, that doesn't attach the pageX and pageY coordinates to "ondrag" event.
+		 * All browsers attach the correct page coordinates to the "dragover" event.
+		 * @bug https://bugzilla.mozilla.org/show_bug.cgi?id=505521
+		 */
+		document.addEventListener("dragover", this.onDragOver);
+	},
+	destroyed() {
+		document.removeEventListener("dragover", this.onDragOver);
+	},
+	methods: {
+		onDragStart(event: DragEvent): void {
+			const { pageX: x, pageY: y } = event;
+
+			this.$emit('dragstart', event);
+
+			if (event.dataTransfer) {
+				event.dataTransfer.setDragImage(this.$refs.draggableDataTransfer as Element, 0, 0);
+			}
+
+			this.$data.dragging = true;
+			this.$data.draggablePosition = { x, y };
```

Comment:
as stated before, would recommend replacing `this.$data.key` with `this.key`
<img width="325" alt="Screenshot 2022-04-14 at 14 35 48" src="https://user-images.githubusercontent.com/4711238/163391754-914d2252-cefa-4a6d-9191-c32a575843fc.png">
<img width="328" alt="Screenshot 2022-04-14 at 14 35 58" src="https://user-images.githubusercontent.com/4711238/163391764-7dfa5f96-011c-4d53-a81e-4f1f3e3c8bc5.png">

Reply:
Thank you for pointing that out. I forgot to refactor it. Changed to `this.key`.